hedera-node/src/main/java/com/hedera/services/store/schedule/HederaScheduleStore.java

```diff
@@ -183,49 +258,239 @@ public boolean isCreationPending() {
+
+			var bySecond = schedules.get().byExpirationSecond().get(secondKey);
+
+			if (bySecond != null) {
+				bySecond = bySecond.asWritable();
+				bySecond.removeId(existingSchedule.calculatedExpirationTime(), idToDelete.getKeyAsLong());
+
+				if (bySecond.getIds().isEmpty()) {
+					schedules.get().byExpirationSecond().remove(secondKey);
+				} else {
+					schedules.get().byExpirationSecond().put(secondKey, bySecond);
+				}
+			}
+
+
+			var equalityKey = new ScheduleEqualityVirtualKey(existingSchedule.equalityCheckKey());
+
+			var byEquality = schedules.get().byEquality().get(equalityKey);
+
+			if (byEquality != null) {
+				byEquality = byEquality.asWritable();
+				byEquality.remove(existingSchedule.equalityCheckValue(), idToDelete.getKeyAsLong());
+
+				if (byEquality.getIds().isEmpty()) {
+					schedules.get().byEquality().remove(equalityKey);
+				} else {
+					schedules.get().byEquality().put(equalityKey, byEquality);
+				}
+			}
+		}
+	}
+
+	@Override
+	public boolean advanceCurrentMinSecond(Instant consensusTime) {
+
+		boolean changed = false;
+
+		long curSecond = schedules.get().getCurrentMinSecond();
+
+		while ((consensusTime.getEpochSecond() > curSecond)
+				&& (!schedules.get().byExpirationSecond().containsKey(new SecondSinceEpocVirtualKey(curSecond)))) {
+
+			++curSecond;
+			changed = true;
+		}
+
+		if (changed) {
+			schedules.get().setCurrentMinSecond(curSecond);
+			return true;
+		}
+
+		return false;
+	}
+
+	@Override
+	public List<ScheduleID> nextSchedulesToExpire(Instant consensusTime) {
+
+		long curSecond = schedules.get().getCurrentMinSecond();
+
+		if (!shouldProcessSecond(consensusTime, curSecond)) {
+			return Collections.emptyList();
+		}
+
+		var bySecondKey = new SecondSinceEpocVirtualKey(curSecond);
+
+		var bySecond = schedules.get().byExpirationSecond().get(bySecondKey);
+
+		final List<ScheduleID> list = new ArrayList<>();
+		List<Pair<RichInstant, Long>> toRemove = new ArrayList<>();
+
+		if (bySecond != null) {
+			outer: for (var entry : bySecond.getIds().entrySet()) {
+				var instant = entry.getKey();
+				var ids = entry.getValue();
+				for (int i = 0; i < ids.size(); ++i) {
+					var id = ids.get(i);
+					var scheduleId = EntityNum.fromLong(id).toGrpcScheduleId();
+
+					var schedule = getNoError(scheduleId);
+
+					if (schedule == null) {
+						log.error("bySecond contained a schedule that does not exist! Removing it! second={}, id={}",
+								curSecond, scheduleId);
+						toRemove.add(Pair.of(instant, id));
+
+					} else if (schedule.calculatedExpirationTime().getSeconds() != curSecond) {
+						log.error("bySecond contained a schedule in the wrong spot! Removing and expiring it! spot={}, id={}, schedule={}",
+								curSecond, scheduleId, schedule);
+						toRemove.add(Pair.of(instant, id));
+						list.add(scheduleId);
+
+					} else if (schedule.isDeleted() || schedule.isExecuted()) {
+						list.add(scheduleId);
+
+					} else {
+						break outer;
```

Comment:
Using a labeled `break` seems confusing, the semantics would be the same with just a `break` here afaict.